### PR TITLE
Update Specification Workflow

### PR DESCRIPTION
**Pull request by gmourier**
# Summary

Rewrites and simplifies the readme to bring it up to date with the [new specification writing and release process](https://www.notion.so/meilisearch/Specification-process-b017a397baaf492c9bc375bcca7ca4c6).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,128 +1,78 @@
...
-Once the file is created with the two parts, `Summary` and `Motivation` filled in. It is time to create a draft PR. The PR should remain in draft state until the content is, according to the author, ready for comments. However, outsiders can make some recommendations, but these are more in the spirit of helping rather than suggesting changes.
-
-### Open State
-
-Once the specification is ready for comments in the owner's eyes, the owner can then switch the PR to open. At that time, it can be commented on, modified, challenged by these peers. If there is the slightest friction during this process, a discussion will be recommended offline within Meilisearch. The PR owner will organize this discussion, and at the end of this interview, the owner will make any necessary changes.
-
-At this step, the PR should be primarily tagged as `In Progress`.
-
-Note that if the pull request does not satisfy Open State conditions it will go back to the previous step above (***Draft State***).
+To start a new specification, a new branch must start from `main` or from `release-vX.X.X` if the changes to write are already planned for an upcoming release.
```
[the remark this answers is not in the document]
Comment (curquiza) [reply README.md:7]:
```suggestion
To start a new specification, a new branch must start
- from `release-vX.X.X` if the related changes are already planned for the release `vX.X.X`
- from `main` if you don't know in which release the changes will be integrated
```

File: README.md [hunk 2]
```diff
@@ -1,128 +1,78 @@
...
-To validate and merge the PR, it must be validated by at least three people.
+The [pull-request template](pull_request_template.md) must be filled in when the pull-request is created.
 
-- One person from the Integration team (preferably @curquiza).
-- One person from the Core team (preferably @Kerollmops).
-- One person from the DevRel team (preferably the person in charge of the documentation).
+## Review State
 
-Note that if the specification does not satisfy the merging conditions, it will go back to the previous step above (***Open State***) since it's lacking some pieces of information in the reviewer's eyes.
+Once the specification is ready for review in the owner's eyes, the owner can then switch the PR to open.
```
Comment (curquiza) [inline README.md:17]:
```suggestion
It's up to the maintainers of this repository to decide when the PR is ready to be reviewed and which persons should review it.
```

File: README.md [hunk 3]
```diff
@@ -1,128 +1,78 @@
...
-Once it's done, the specification is accepted and merged to the main branch.
+To be merged, a specification pull-request should follow the given rules:
 
-## Specification Revision
-
-Any Meilisearch's specification can change over time due to diverse factors. Sometimes related to change made on another specification concerning core engine code that could impact it, or simply product enhancement.
-
-In this precise case, a new PR is created and should follow the specification workflow described above.
+- It must be approved as described in the [Review State](#review-state) section.
+- It must be branched on a given `release-vX.X.X` branch.
```
Comment (curquiza) [inline README.md:32]:
```suggestion
- The PR must point to the right `release-vX.X.X` branch.
```

File: README.md [hunk 4]
```diff
@@ -1,128 +1,78 @@
...
-#### I. Summary
-
-One paragraph explaining the feature from the perspective of a user, e.g. "When I do X I want Y to happen because Z". This paragraph describes the problem and solution simply, without going into detail.
+# Release Worfklow
 
-#### II. Motivation
+The following steps should happens the day a Meilisearch release is shipped:
 
-Why develop this feature? What use cases does it support? What is the expected outcome? Add links to related issues and discussions.
+- Pull-requests describing changes for a release are merged and squashed into the corresponding `release-vX.X.X` branch.
```
Comment (curquiza) [inline README.md:46]:
```suggestion
- Pull-requests describing changes for a release are squashed and merged into the corresponding `release-vX.X.X` branch.
```

File: README.md [hunk 5]
```diff
@@ -1,128 +1,78 @@
...
+- `release-vX.X.X` is merged and squashed into `main`.
```
Comment (curquiza) [inline README.md:47]:
```suggestion
- `release-vX.X.X` is squashed and merged into `main`.
```

File: README.md [hunk 6]
```diff
@@ -1,128 +1,78 @@
...
+### 1. Summary
 
-If the feature requires additions or updates to the documentation, they should be noted here. It's the role of the documentation team to ensure this section of the feature specification is accurate.
+Summarize the specification with a short paragraph.
 
-#### VI. Impact on SDKs
+### 2. Motivation
 
-If the feature requires additions or updates to the SDks, they should be noted here. It's the role of the integration team to ensure this section of the feature specification is accurate.
+Explains what use cases does it support.
```
Comment (curquiza) [inline README.md:62]:
```suggestion
Explain which use cases are supported.
```

File: README.md [hunk 7]
```diff
@@ -1,128 +1,78 @@
...
-### 2. Technical Aspects
+### 3. Functional Specification
 
-This section of the feature specification document has a much narrower audience: the developer(s) that will implement the feature. Its goal is to discuss and clarify how the feature will be developed and implemented when its not a trivial concern.
+This section gives a high level overview of the feature. It should avoid technical language so that it can be understood by a general audience (think user-level).
 
-> Internal technical teams may use this part to keep track of technical brainstorms, related proto code or basecode examples to clear the technical details of the specification.
+- Describes the API resource and endpoints. (Methods, URL, query parameters, body definition, status code).
+- Explains the feature through examples.
+- Lists error cases.
```
Comment (curquiza) [inline README.md:70]:
```suggestion
- Describe the API resource and endpoints. (Methods, URL, query parameters, body definition, status code).
- Explain the feature through examples.
- List error cases.
```

Comment (curquiza) [reply README.md:70]: For me you are talking to the person reading the process, so you give information about what to do

**Review by Kerollmops — APPROVED** [review]
Looks good to me! Thank you, much simpler.

**Review by curquiza — COMMENTED** [review]
🎉 !!!!

**Review by curquiza — APPROVED** [review]

**Pull request merged**